From ad03701dff464b3c81253ebe50bc6a3557e25578 Mon Sep 17 00:00:00 2001
From: Subbaraya Sundeep <sbhatta@marvell.com>
Date: Thu, 9 May 2019 18:26:41 +0530
Subject: [PATCH 225/386] octeontx2-pf: Cleanup properly during driver remove

When unbinding a PF driver sriov needs to be disabled
before disabling PF interrupts. Otherwise all the VF cleanup
messages never get response from AF since PF interrupt is
disabled. Also VF ntuple filters installed by a PF needs to
be deleted before VF removal otherwise already running
traffic may hit VF ntuple filter rules for removed VFs.

Change-Id: I1e7cefde646dc88ef5b14387e00a709b7f2d8876
Signed-off-by: Subbaraya Sundeep <sbhatta@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/8803
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
[RH: Original patch taken from marvell 88F3720 board support SDK 10.0-PR2003]
Signed-off-by: Ruiqiang Hao <Ruiqiang.Hao@windriver.com>
---
 drivers/net/ethernet/marvell/octeontx2/nic/otx2_pf.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/drivers/net/ethernet/marvell/octeontx2/nic/otx2_pf.c b/drivers/net/ethernet/marvell/octeontx2/nic/otx2_pf.c
index 9625519927fe..061855003a6e 100644
--- a/drivers/net/ethernet/marvell/octeontx2/nic/otx2_pf.c
+++ b/drivers/net/ethernet/marvell/octeontx2/nic/otx2_pf.c
@@ -2412,6 +2412,7 @@ static int otx2_sriov_disable(struct pci_dev *pdev)
 	if (!pci_num_vf(pdev))
 		return 0;
 
+	otx2_delete_vf_ethtool_flows(pf);
 	pci_disable_sriov(pdev);
 
 	for (i = 0; i < pci_num_vf(pdev); i++)
@@ -2422,7 +2423,6 @@ static int otx2_sriov_disable(struct pci_dev *pdev)
 	otx2_flr_wq_destroy(pf);
 	otx2_disable_pfvf_mbox_intr(pf);
 	otx2_pfvf_mbox_destroy(pf);
-	otx2_delete_vf_ethtool_flows(pf);
 
 	return 0;
 }
@@ -2454,14 +2454,14 @@ static void otx2_remove(struct pci_dev *pdev)
 	otx2_cgx_config_linkevents(pf, false);
 
 	unregister_netdev(netdev);
+
+	otx2_sriov_disable(pf->pdev);
 	otx2_ptp_destroy(pf);
 	otx2_destroy_ethtool_flows(pf);
 
-	otx2_disable_mbox_intr(pf);
-
 	otx2_detach_resources(&pf->mbox);
+	otx2_disable_mbox_intr(pf);
 	otx2_pfaf_mbox_destroy(pf);
-	otx2_sriov_disable(pf->pdev);
 	pci_free_irq_vectors(pf->pdev);
 	pci_set_drvdata(pdev, NULL);
 	free_netdev(netdev);
-- 
2.17.1

